keylime/ima.py

```diff
@@ -349,6 +355,11 @@ def read_allowlist(al_path=None, checksum="", gpg_sig_file=None, gpg_key_file=No
         # version 2 added 'keyrings'
         if "keyrings" not in alist:
             alist["keyrings"] = {}
+        # version 3 added 'ima"
+        if "ima" not in alist:
+            alist["ima"] = {
+                "ignored_keyrings": []
+            }
```

Comment:
I added this now because I had added it for v2 as well. However, I am not sure whether this is correct or necessary because if the policy comes in via `process_allowlist`, which is the case with some tests, then this type of extension is not done (a bug?) and then it's easier to do what I did above with `allowlist.get('ima', {}).get('ignored_keyrings', [])` to handle optional fields added only later on.

Reply:
The tenant first runs `read_allowlist` and then with that output `process_allowlist` which not all tests do (that could be a bug). The verifier just uses the JSON send via the API. 

I would add a function that adds necessary fields with sane defaults and run it on the JSON provided via the API.